core/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -472,6 +473,24 @@ class DeltaTable private[tables](
   def upgradeTableProtocol(readerVersion: Int, writerVersion: Int): Unit = {
     deltaLog.upgradeProtocol(Protocol(readerVersion, writerVersion))
   }
+
+  /**
+   * Restores delta table to specific version.
+   * @param version version to be restored
+   * @since 1.2.0
+   */
+  def restore(version: Long): Unit = {
```

Comment:
My suggestion to separate these overloaded methods with 1 param each is to name them different. Say `restoreToVersion` and `restoreToTimestamp`